### PR TITLE
TestPullNonExistingImage: Simplify and move to integration

### PR DESCRIPTION
**Pull request by vvoland**
**- What I did**

Rewrote `TestPullNonExistingImage` to use parallel subtests instead of the custom goroutines solution.
Deduplicated asserts and rewrote with `gotest.tools`.

**- How to verify it**
See `TestImagePullNonExisting`

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: integration/image/pull_test.go [hunk 1]
```diff
@@ -155,3 +158,48 @@ func TestImagePullStoredfDigestForOtherRepo(t *testing.T) {
...
+
+// TestImagePullNonExisting pulls non-existing images from the central registry, with different
+// combinations of implicit tag and library prefix.
+func TestImagePullNonExisting(t *testing.T) {
+	ctx := setupTest(t)
+
+	for _, tc := range []struct {
+		alias string
+		tag   string
+	}{
```
Comment (rumpl) [inline integration/image/pull_test.go:170]: This struct could have been “ref” and “all”

**Review by rumpl — APPROVED** [review]
LGTM

I had the same change in some branch/stash :)

File: integration/image/pull_test.go (hunk 1 above)
Comment (vvoland) [reply integration/image/pull_test.go]: Actually `ref` alone is enough here. Done.

**Review by laurazard — APPROVED** [review]
LGTM

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**